plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestHiveFileIterator.java

```diff
@@ -30,6 +30,9 @@ public void testRelativeHiddenPathDetection()
         String root = new Path("file:///root-path").toUri().getPath();
         assertTrue(isHiddenOrWithinHiddenParentDirectory(new Path(root, ".hidden/child"), root));
         assertTrue(isHiddenOrWithinHiddenParentDirectory(new Path(root, "_hidden.txt"), root));
+        String rootWithSlash = new Path("file:///root-path/").toUri().getPath();
```

Comment:
Ideally, we should add query based tests than unit tests. Can you try adding that? No need to remove this test.

Reply:
@ebyhr Can you give me an example of a query test?

Reply:
@ebyhr in this case I would suggest that a query test isn’t going to be all that useful since the behavior of file listings is somewhat dependent on the contents of existing table location metadata (which may have been created or validated differently) and file system implementation details which a query test won’t fully cover.

With @dprophet’s changes we can handle paths both with and without trailing slashes correctly, so I don’t think we should block merging the bug fix on adding a query test in this case.